### ISSUE TITLE
Lightweight core library

### ISSUE THREAD

**Issue by sdispater** [python-poetry/poetry#1993]
Currently, to build a Poetry project via the `PEP-517` build mechanism, you need this section:

```toml
[build-system]
requires = ["poetry>=1.0.3"]
build-backend = "poetry.masonry.api"
```

This works but this has the undesirable side-effect of pulling the full Poetry stack and its dependencies. Most of them are not needed to build the project and expose the metadata.

To prevent that, we need to extract parts of the Poetry stack into a more lightweight library with the sole purpose of building projects.

Ideally, this core library should not have any dependency to avoid errors to to circular build dependencies when using `pip install --no-binary :all:`.

**Comment by pradyunsg** [comment]
Just to be sure, this core would be analogous to `flit_core`, correct?

**Comment by sdispater** [comment]
@pradyunsg Yes, the idea is to not require the full Poetry stack just to build projects.

And it will help fix the issue with `pip` when there are cyclic dependencies between the build backend and its own dependencies.

**Comment by ulgens** [comment]
@sdispater Isn't this resolved by #2212 ? Not sure why this one is not automatically closed. Also, we can mark it as done on #1856 .
